### ISSUE TITLE
Incorrect reporting for adding a field to a private struct

### ISSUE THREAD

**Issue by theduke** [rust-dev-tools/rust-semverver#112]
I've just run `cargo semver` on a project where a field was added to a private struct.

This results in the following, somewhat weird error:

```
error: breaking changes in `ContextBuilder`
   --> /home/theduke/dev/github.com/theduke/quickjs-rs/src/lib.rs:117:1
    |
117 | / pub struct ContextBuilder {
118 | |     memory_limit: Option<usize>,
119 | |     console_backend: Option<Box<dyn console::ConsoleBackend>>,
120 | | }
    | |_^
    |
warning: variant field added to variant with private fields (breaking)
   --> /home/theduke/dev/github.com/theduke/quickjs-rs/src/lib.rs:119:5
    |
119 |     console_backend: Option<Box<dyn console::ConsoleBackend>>,
```

This is wrong on two counts:

* there is no enum involved, so the mention of `variant` is incorrect
* adding a private field to a struct is not a breaking change

I guess this will be related to recent rust refactoring.

**Comment by ibabushkin** [comment]
The mention of a `variant` is basically due to an implementation detail, I suppose that can be adjusted (but basically at the level we perform our analysis, structs are essentially just single-variant enums).

Regarding the breakage, I agree, that is a corner case that's incorrectly handled. Oonly adding a first private field would be breaking. I'll get a fix underway.

**Comment by ibabushkin** [comment]
Fixed in 0.1.45, available on crates.io. Thanks for reporting!

**Comment by theduke** [comment]
:+1:
